* python3-dlib linking issue #yocto #zeus #python
@ 2021-05-17 20:11 Bel Hadj Salem Talel
  0 siblings, 0 replies; only message in thread
From: Bel Hadj Salem Talel @ 2021-05-17 20:11 UTC (permalink / raw)
  To: yocto

[-- Attachment #1: Type: text/plain, Size: 2065 bytes --]

Hi All,
I'm trying to bitbake a recipe for python3-dlib (the python API).
Here is the recipe:

SUMMARY = "A toolkit for making real world machine learning and data analysis applications"
HOMEPAGE = "https://github.com/davisking/dlib"
LICENSE = "Boost-Software"
LIC_FILES_CHKSUM = "file://dlib/LICENSE.txt;md5=2c7a3fa82e66676005cd4ee2608fd7d2 \
file://dlib/external/pybind11/LICENSE;md5=beb87117af69fd10fbf9fb14c22a2e62 \
file://dlib/external/libpng/LICENSE;md5=243135ddedf702158f9170807cbcfb66 \
file://docs/docs/license.xml;md5=2e6ff4080dcb217d4d300b90e9aabb5b \
file://examples/LICENSE_FOR_EXAMPLE_PROGRAMS.txt;md5=57eee82829ed297e23d84de5f905afee \
file://examples/video_frames/license.txt;md5=127ee508b60a6be9dea8af3b441993dc \
file://python_examples/LICENSE_FOR_EXAMPLE_PROGRAMS.txt;md5=064f53ab40ea2b6a4bba1324149e4fde"

SRC_URI = "git://github.com/davisking/dlib.git;protocol=https"
PV = "1.0+git${SRCPV}"
SRCREV = "3b794540baeabbcd033b544230401967106d5483"
S = "${WORKDIR}/git"
inherit setuptools3
DEPENDS += "python3 cmake-native"
FILES_${PN}-dev += "${libdir}/cmake/dlib"
EXTRA_OECMAKE += "-DDLIB_NO_GUI_SUPPORT=ON -DBUILD_SHARED_LIBS=ON -DDLIB_USE_CUDA=OFF"
------

When I try to bitbake it, I get a linking issue with ld telling me that libnsl.so in wrong format:

[100%] Linking CXX shared module /media/talel/data/menzu-zeus/ttdemo/tmp/work/aarch64-poky-linux/python3-dlib/1.0+gitAUTOINC+3b794540ba-r0/git/build/lib.linux-x86_64-3.7/dlib.cpython-37m-aarch64-linux-gnu.so
| /media/talel/data/menzu-zeus/ttdemo/tmp/work/aarch64-poky-linux/python3-dlib/1.0+gitAUTOINC+3b794540ba-r0/recipe-sysroot-native/usr/bin/aarch64-poky-linux/../../libexec/aarch64-poky-linux/gcc/aarch64-poky-linux/9.2.0/ld: /media/talel/data/menzu-zeus/ttdemo/tmp/work/aarch64-poky-linux/python3-dlib/1.0+gitAUTOINC+3b794540ba-r0/recipe-sysroot-native/usr/lib/libnsl.so: error adding symbols: file in wrong format
| collect2: error: ld returned 1 exit status

Did anyone make it work? Or does anyone know how to solve the problem?

Thanks,
Talel

[-- Attachment #2: Type: text/html, Size: 2795 bytes --]

^ permalink raw reply	[flat|nested] only message in thread
only message in thread, other threads:[~2021-05-17 20:11 UTC | newest]

Thread overview: (only message) (download: mbox.gz / follow: Atom feed)
-- links below jump to the message on this page --
2021-05-17 20:11 python3-dlib linking issue #yocto #zeus #python Bel Hadj Salem Talel

